### PR TITLE
Adapt SVGLengthList to correct API structure, write subpages

### PR DESCRIPTION
**Pull request by Josh-Cena**
This PR writes the pages for `SVGLengthList` because the landing page is not the correct structure. Once accepted we can use the same format for the other SVG lists.

### CODE REVIEW

File: files/en-us/web/api/svglengthlist/index.md [hunk 1]
```diff
@@ -7,297 +7,112 @@ browser-compat: api.SVGLengthList
 
 {{APIRef("SVG")}}
 
-## SVG length list interface
-
-The `SVGLengthList` defines a list of {{ domxref("SVGLength") }} objects.
+The `SVGLengthList` defines a list of {{ domxref("SVGLength") }} objects. It is used for the {{domxref("SVGAnimatedLengthList.baseVal", "baseVal")}} and {{domxref("SVGAnimatedLengthList.animVal", "animVal")}} properties of {{domxref("SVGAnimatedLengthList")}}.
```
Comment (sideshowbarker) [inline files/en-us/web/api/svglengthlist/index.md:10]:
```suggestion
The `SVGLengthList` interface defines a list of {{ domxref("SVGLength") }} objects. It is used for the {{domxref("SVGAnimatedLengthList.baseVal", "baseVal")}} and {{domxref("SVGAnimatedLengthList.animVal", "animVal")}} properties of {{domxref("SVGAnimatedLengthList")}}.
```

File: files/en-us/web/api/svglengthlist/index.md [hunk 2]
```diff
@@ -7,297 +7,112 @@ browser-compat: api.SVGLengthList
...
 
 An `SVGLengthList` object can be designated as read only, which means that attempts to modify the object will result in an exception being thrown.
 
 An `SVGLengthList` is indexable and can be accessed like an array.
```
Comment (sideshowbarker) [inline files/en-us/web/api/svglengthlist/index.md:14]:
```suggestion
An `SVGLengthList` object is indexable and can be accessed like an array.
```

File: files/en-us/web/api/svglengthlist/index.md [hunk 3]
```diff
@@ -7,297 +7,112 @@ browser-compat: api.SVGLengthList
...
+- {{domxref("SVGLengthList.removeItem", "removeItem()")}}
+  - : Removes an existing item from the list.
+- {{domxref("SVGLengthList.replaceItem", "replaceItem()")}}
+  - : Replaces an existing item in the list with a new item.
+
+## Examples
+
+### Using SVGLengthList
+
+`SVGLengthList` can be retrived from {{domxref("SVGAnimatedLengthList")}}, which itself is retrievable from many animatable length attributes, such as {{domxref("SVGTextPositioningElement.x")}}.
```
Comment (sideshowbarker) [inline files/en-us/web/api/svglengthlist/index.md:44]:
```suggestion
An `SVGLengthList` object can be retrived from {{domxref("SVGAnimatedLengthList")}}, which itself is retrievable from many animatable length attributes, such as {{domxref("SVGTextPositioningElement.x")}}.
```

**Review by sideshowbarker — APPROVED** [review]

File: files/en-us/web/api/svglengthlist/appenditem/index.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+The {{domxref("SVGLength")}} that was added to the list.
+
+### Exceptions
+
+- {{domxref("DOMException")}} `NoModificationAllowedError`
+  - : Thrown when the list is read-only.
+
+## Examples
+
+See {{domxref("SVGLengthList")}} a complete example.
```
Comment (Josh-Cena) [inline files/en-us/web/api/svglengthlist/appenditem/index.md:35]:
```suggestion
See {{domxref("SVGLengthList")}} for a complete example.
```

File: files/en-us/web/api/svglengthlist/clear/index.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+None ({{jsxref("undefined")}}).
+
+### Exceptions
+
+- {{domxref("DOMException")}} `NoModificationAllowedError`
+  - : Thrown when the list is read-only.
+
+## Examples
+
+See {{domxref("SVGLengthList")}} a complete example.
```
Comment (Josh-Cena) [inline files/en-us/web/api/svglengthlist/clear/index.md:34]:
```suggestion
See {{domxref("SVGLengthList")}} for a complete example.
```

File: files/en-us/web/api/svglengthlist/clear/index.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+title: "SVGLengthList: clear() method"
+short-title: clear()
+slug: Web/API/SVGLengthList/clear
+page-type: web-api-instance-method
+browser-compat: api.SVGLengthList.clear
+---
+
+{{APIRef("SVG")}}
+
+The **`clear()`** method of the {{domxref("SVGLengthList")}} interface clears all existing current items from the list, with the result being an empty list.
```
Comment (Josh-Cena) [inline files/en-us/web/api/svglengthlist/clear/index.md:11]:
```suggestion
The **`clear()`** method of the {{domxref("SVGLengthList")}} interface clears all existing items from the list, with the result being an empty list.
```

File: files/en-us/web/api/svglengthlist/getitem/index.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+The {{domxref("SVGLength")}} at the specified index in the list.
+
+### Exceptions
+
+- {{domxref("DOMException")}} `IndexSizeError`
+  - : Thrown when the index is out of bounds for the list.
+
+## Examples
+
+See {{domxref("SVGLengthList")}} a complete example.
```
Comment (Josh-Cena) [inline files/en-us/web/api/svglengthlist/getitem/index.md:35]:
```suggestion
See {{domxref("SVGLengthList")}} for a complete example.
```

File: files/en-us/web/api/svglengthlist/getitem/index.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+title: "SVGLengthList: getItem() method"
+short-title: getItem()
+slug: Web/API/SVGLengthList/getItem
+page-type: web-api-instance-method
+browser-compat: api.SVGLengthList.getItem
+---
+
+{{APIRef("SVG")}}
+
+The **`getItem()`** method of the {{domxref("SVGLengthList")}} interface returns the specified item from the list. The returned item is the item itself and not a copy. Any changes made to the item are immediately reflected in the list. The first item is number 0.
```
Comment (Josh-Cena) [inline files/en-us/web/api/svglengthlist/getitem/index.md:11]:
```suggestion
The **`getItem()`** method of the {{domxref("SVGLengthList")}} interface returns the specified item from the list. The returned item is the item itself and not a copy. Any changes made to the item are immediately reflected in the list. The first item is indexed 0.
```

File: files/en-us/web/api/svglengthlist/initialize/index.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+title: "SVGLengthList: initialize() method"
+short-title: initialize()
+slug: Web/API/SVGLengthList/initialize
+page-type: web-api-instance-method
+browser-compat: api.SVGLengthList.initialize
+---
+
+{{APIRef("SVG")}}
+
+The **`initialize()`** method of the {{domxref("SVGLengthList")}} interface clears all existing current items from the list and re-initializes the list to hold the single item specified by the parameter. If the inserted item is already in a list, it is removed from its previous list before it is inserted into this list. The inserted item is the item itself and not a copy. The return value is the item inserted into the list.
```
Comment (Josh-Cena) [inline files/en-us/web/api/svglengthlist/initialize/index.md:11]:
```suggestion
The **`initialize()`** method of the {{domxref("SVGLengthList")}} interface clears all existing items from the list and re-initializes the list to hold the single item specified by the parameter. If the inserted item is already in a list, it is removed from its previous list before it is inserted into this list. The inserted item is the item itself and not a copy. The return value is the item inserted into the list.
```

File: files/en-us/web/api/svglengthlist/initialize/index.md [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+The {{domxref("SVGLength")}} that was added to the list.
+
+### Exceptions
+
+- {{domxref("DOMException")}} `NoModificationAllowedError`
+  - : Thrown when the list is read-only.
+
+## Examples
+
+See {{domxref("SVGLengthList")}} a complete example.
```
Comment (Josh-Cena) [inline files/en-us/web/api/svglengthlist/initialize/index.md:35]:
```suggestion
See {{domxref("SVGLengthList")}} for a complete example.
```

File: files/en-us/web/api/svglengthlist/insertitembefore/index.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+title: "SVGLengthList: insertItemBefore() method"
+short-title: insertItemBefore()
+slug: Web/API/SVGLengthList/insertItemBefore
+page-type: web-api-instance-method
+browser-compat: api.SVGLengthList.insertItemBefore
+---
+
+{{APIRef("SVG")}}
+
+The **`insertItemBefore()`** method of the {{domxref("SVGLengthList")}} interface inserts a new item into the list at the specified position. The first item is number 0. If the new item is already in a list, it is removed from its previous list before it is inserted into this list. The inserted item is the item itself and not a copy. If the item is already in this list, note that the index of the item to insert before is before the removal of the item. If the index is equal to 0, then the new item is inserted at the front of the list. If the index is greater than or equal to the {{domxref("SVGLengthList.length", "length")}}, then the new item is appended to the end of the list.
```
Comment (Josh-Cena) [inline files/en-us/web/api/svglengthlist/insertitembefore/index.md:11]:
```suggestion
The **`insertItemBefore()`** method of the {{domxref("SVGLengthList")}} interface inserts a new item into the list at the specified position. The first item is indexed 0. If the new item is already in a list, it is removed from its previous list before it is inserted into this list. The inserted item is the item itself and not a copy. If the item is already in this list, note that the index of the item to insert before is before the removal of the item. If the index is equal to 0, then the new item is inserted at the front of the list. If the index is greater than or equal to the {{domxref("SVGLengthList.length", "length")}}, then the new item is appended to the end of the list.
```

File: files/en-us/web/api/svglengthlist/insertitembefore/index.md [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+The {{domxref("SVGLength")}} that was added to the list.
+
+### Exceptions
+
+- {{domxref("DOMException")}} `NoModificationAllowedError`
+  - : Thrown when the list is read-only.
+
+## Examples
+
+See {{domxref("SVGLengthList")}} a complete example.
```
Comment (Josh-Cena) [inline files/en-us/web/api/svglengthlist/insertitembefore/index.md:37]:
```suggestion
See {{domxref("SVGLengthList")}} for a complete example.
```

File: files/en-us/web/api/svglengthlist/removeitem/index.md [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+### Exceptions
+
+- {{domxref("DOMException")}} `NoModificationAllowedError`
+  - : Thrown when the list is read-only.
+- {{domxref("DOMException")}} `IndexSizeError`
+  - : Thrown when the index is out of bounds for the list.
+
+## Examples
+
+See {{domxref("SVGLengthList")}} a complete example.
```
Comment (Josh-Cena) [inline files/en-us/web/api/svglengthlist/removeitem/index.md:37]:
```suggestion
See {{domxref("SVGLengthList")}} for a complete example.
```

File: files/en-us/web/api/svglengthlist/replaceitem/index.md [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+### Exceptions
+
+- {{domxref("DOMException")}} `NoModificationAllowedError`
+  - : Thrown when the list is read-only.
+- {{domxref("DOMException")}} `IndexSizeError`
+  - : Thrown when the index is out of bounds for the list.
+
+## Examples
+
+See {{domxref("SVGLengthList")}} a complete example.
```
Comment (Josh-Cena) [inline files/en-us/web/api/svglengthlist/replaceitem/index.md:39]:
```suggestion
See {{domxref("SVGLengthList")}} for a complete example.
```

File: files/en-us/web/api/svglengthlist/index.md [hunk 15]
```diff
@@ -7,297 +7,112 @@ browser-compat: api.SVGLengthList
...
-          </li>
-        </ul>
-      </td>
-    </tr>
-  </tbody>
-</table>
+- {{domxref("SVGLengthList.appendItem", "appendItem()")}}
+  - : Inserts a new item at the end of the list.
+- {{domxref("SVGLengthList.clear", "clear()")}}
+  - : Clears all existing current items from the list, with the result being an empty list.
```
Comment (Josh-Cena) [inline files/en-us/web/api/svglengthlist/index.md:28]:
```suggestion
  - : Clears all existing items from the list, with the result being an empty list.
```

File: files/en-us/web/api/svglengthlist/index.md [hunk 16]
```diff
@@ -7,297 +7,112 @@ browser-compat: api.SVGLengthList
...
+- {{domxref("SVGLengthList.initialize", "initialize()")}}
+  - : Clears all existing current items from the list and re-initializes the list to hold the single item specified by the parameter.
```
Comment (Josh-Cena) [inline files/en-us/web/api/svglengthlist/index.md:30]:
```suggestion
  - : Clears all existing items from the list and re-initializes the list to hold the single item specified by the parameter.
```

**Comment by Josh-Cena** [comment]
Thanks @sideshowbarker for the review!

**Pull request merged**